addons/OPT/GPS/fn_clientInitEH.sqf

```diff
@@ -0,0 +1,134 @@
+*
+* Server Only:
+* No
+* 
+* Global:
+* No
+* 
+* API:
+* No
+* 
+* Example:
+* call FUNC(clientInitEH);
+*/
+
+#include "macros.hpp";
+
+["Killed", {
+    DUMP("KILLED");
+    DUMP(_this);
+    params ["_unit", "_killer", "_instigator", "_useEffects"];
+    _unit = _unit select 0;
+    DUMP(_unit);
+    DUMP(group _unit);
+    DUMP(group CLib_Player);
+    DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+    [[str _unit], QFUNC(removeUnitFromGPS), -2] call CFUNC(remoteExec); // Trigger everywhere except server
+    DUMP(format ["ICON REMOVED BECAUSE KILLED: %1", _unit]);
+    DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+
+}] call CFUNC(addEventhandler);
+
+
+["playerChanged", {
+    params ["_data", "_params"];
+    _data params ["_currentPlayer", "_oldPlayer"];
+
+    [{
+        private _currentPlayer = _this;
+        DUMP(_currentPlayer);
+        DUMP(alive _currentPlayer);
+        DUMP(alive CLib_Player);
+        DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+        [[_currentPlayer], QFUNC(addUnitToGPS), -2] call CFUNC(remoteExec); // Trigger everywhere except server
+        DUMP(format ["ICON ADDED BECAUSE PLAYERCHANGED: %1", _currentPlayer]);
+        DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+    }, _currentPlayer] call CLib_fnc_execNextFrame;
+}] call CFUNC(addEventhandler);
+
+// ["Respawn", {
+//     DUMP("RESPAWN");
+//     DUMP(_this);
+//     params ["_data", "_params"];
+//     _data params ["_unit"];
+//     DUMP(_unit);
+//     DUMP(alive _unit);
+//     DUMP(alive CLib_Player);
+//     {
+//         DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+//         [[_unit], QFUNC(addUnitToGPS), -2] call CFUNC(remoteExec); // Trigger everywhere except server
+//         DUMP("ICON ADDED BECAUSE RESPAWN: " + _iconId);
+//         DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+//     } call CFUNC(execNextFrame);
+// }, [player]] call CFUNC(addEventhandler);
+
+
+["playerJoined", {
+    DUMP("PLAYER JOINED");
+    DUMP(_this);
+    params ["_unit", ""];
+    DUMP(_unit);
+    DUMP(group _unit);
+    DUMP(group CLib_Player);
+    DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+    [[_unit], QFUNC(addUnitToGPS), -2] call CFUNC(remoteExec); // Trigger everywhere except server
+    DUMP(format ["ICON ADDED BECAUSE PLAYER JOINED: %1", _unit]);
+    DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+}] call CFUNC(addEventhandler);
+
+
+// Revive Handlers
+["opt_unitUnconscious", {
```

Comment:
Wo wird dieses Event abgesetzt? Würde nämlich schwer vorschlagen für Eventnamen ebenfalls Makro-Konstanten zu vergeben. Außerdem würde ich in den tatsächlichen Namen noch iwo den Substring `Event` mit rein bringen. dazu sollte dieses Event sowohl beim bewusstlos werden, als auch beim aufwachen abgesetzt werden, wobei ein weiteres Argument dem jeweiligen EH sagt, was gerade der Fall ist.

Analoges gilt für alle folgenden Events